File: src/test/java/kitchenpos/helper/ProductHelper.java
package kitchenpos.helper;


import kitchenpos.domain.Product;

import java.math.BigDecimal;
import java.util.UUID;

public final class ProductHelper {

    private static final String DEFAULT_NAME = "테스트 기본 상품명";

    private ProductHelper() {
    }

    public static Product create(BigDecimal price) {
        return create(DEFAULT_NAME, price);
    }

    public static Product create(String name, BigDecimal price) {
        Product product = new Product();
        product.setId(UUID.randomUUID());
        product.setName(name);
        product.setPrice(price);
        return product;
    }

}

File: src/test/kotlin/kitchenpos/product/ProductHelper.kt
package kitchenpos.product

import io.restassured.RestAssured
import kitchenpos.domain.Product


class ProductHelper {
    companion object {
        fun 상품이름으로_상품_조회(name: String): Product {
            return RestAssured
                .get("/api/products")
                .then().log().all().extract().`as`(Array<Product>::class.java)
                .find { it.name == name }!!
        }
    }
}
